Update no-code module workspace API docs, remove TFE exclusion [IPE-812] (#654)
hashicorp/terraform-docs-common: website/docs/cloud-docs/api-docs/no-code-provisioning.mdx

# No-Code Provisioning API

The no-code provisioning API allows you to create, configure, and deploy Terraform modules in the no-code provisioning workflows within HCP Terraform. For more information on no-code modules, see [Designing No-Code Ready Modules](/terraform/cloud-docs/no-code-provisioning/module-design).

## Allow no-code provisioning of a module

`POST /organizations/:organization_name/no-code-modules`

| Parameter            | Description                                                                                                                                                              |
| -------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| `:organization_name` | The name of the organization the module belongs to.  |

> To deploy a module using the no-code provisioning workflow, you must meet the following requirement:

Reviewer: ```suggestion
To deploy a module using the no-code provisioning workflow, the module must meet the following requirement:
```
Right? The requirements are for the module? Just to state the obvious, this really should be a separate usage doc with a "Requirements" section, but here we are.

1. The module must exist in your organization's private registry.
1. The module must meet the [design requirements](/terraform/cloud-docs/no-code-provisioning/module-design#requirements) for a no-code module.
1. You must enable no-code provisioning for the module.

> You can use this API to enable no-code provisioning for a specific module. You can also use this API to set options for the allowed values of a variable for a no-code module in your organization.

Reviewer: ```suggestion
You can send a `POST` request to the `/no-code-modules` endpoint to enable no-code provisioning for a specific module. You can also call this endpoint to set options for the allowed values of a variable for a no-code module in your organization.
```
Just to be a little more concrete.

}
```

## Update no-code provisioning settings

`PATCH /no-code-modules/:id`

| Parameter | Description                                         |
| --------- | --------------------------------------------------- |
| `:id`     | The unique identifier for a No-Code-Allowed module. |

> Use this API to update the settings for the no-code provisioning of a module. You can use this API to update the following settings:

Reviewer: ```suggestion
Send a `PATCH` request to the `/no-code-modules/:id` endpoint to update the settings for the no-code provisioning of a module. You can update the following settings:
```
For consistency with the previous suggestion.

Properties without a default value are required.

| Key path | Type | Default | Description |
| --- | --- | --- | --- |
| `data.type` | string | | Must be `"no-code-modules"`. |

> | `data.attributes.version-pin` | string   | (previous value) | The module version to use in no-code provisioning workflows. |

Reviewer: ```suggestion
| `data.attributes.version-pin` | string   | Existing value | The module version to use in no-code provisioning workflows. |
```
Correct? We mean that TF uses the value that is already there if you don't supply this attribute, yeah? Also, I don't see a reason put it in parentheses.

> | `data.attributes.enabled` | boolean   | (previous value) | Set to `true` to enable no-code provisioning workflows, or `false` to disable them. |

Reviewer: ```suggestion
| `data.attributes.enabled` | boolean   | Existing value | Set to `true` to enable no-code provisioning workflows, or `false` to disable them. |
```
There are a few more instances in this table. At a minimum, we should remove the parens

The [API call that enables no-code provisioning for a module](#allow-no-code-provisioning-of-a-module-within-an-organization) returns that module's unique identifier.

| Status  | Response                                          | Reason                                                                |
| ------- | ------------------------------------------------- | --------------------------------------------------------------------- |
| [200][] | [JSON API document][] (`type: "no-code-modules"`) | Successfully read the no-code module.                                 |
| [400][] | [JSON API error object][]                         | Invalid `include` parameter.                                          |
| [404][] | [JSON API error object][]                         | Not found, or the user is unauthorized to perform this action.        |
| [422][] | [JSON API error object][]                         | Malformed request body (e.g., missing attributes, wrong types, etc.). |
| [500][] | [JSON API error object][]                         | Internal system failure.                                              |

### Query Parameters

> This endpoint uses our [standard URL query parameters](/terraform/cloud-docs/api-docs#query-parameters). Remember to percent-encode `[` as `%5B` and `]` as `%5D` if your tooling doesn't automatically encode URLs.

Reviewer: ```suggestion
This endpoint uses our [standard URL query parameters](/terraform/cloud-docs/api-docs#query-parameters). Use HTML URL encoding syntax, such as `%5B` to represent `[` and `%5D` to represent `]`, if your tooling does not automatically encode URLs.
```

> This GET endpoint can optionally return related resources if you request them with [the `include` query parameter](/terraform/cloud-docs/api-docs#inclusion-of-related-resources).

Reviewer: ```suggestion
Terraform returns related resources when you add [the `include` query parameter](/terraform/cloud-docs/api-docs#inclusion-of-related-resources) to the request.
```
Is that any more direct?

}
```

## Create a no-code module workspace

This endpoint creates a workspace from a no-code module. 

`POST /no-code-modules/:id/workspaces`

| Parameter            | Description                                |
| -------------------- | ------------------------------------------ |

> | `:id`                | The id of the no-code module to provision. |

Reviewer: ```suggestion
| `:id`                | The ID of the no-code module to provision. |
```

Properties without a default value are required.

| Key path | Type   | Default | Description |
| --- | --- | --- | ---  |
| `data.type`                                             | string  |           | Must be `"workspaces"`. |

> | `data.attributes.name`                                  | string  |           | The name of the workspace, which can only include letters, numbers, -, and _. This will be used as an identifier and must be unique in the organization. |

Reviewer: ```suggestion
| `data.attributes.name`                                  | string  |           | The name of the workspace. You can only include letters, numbers, `-`, and `_`. Terraform uses this value to identify the workspace and must be unique in the organization. |
```

| `data.attributes.description`                           | string  | (nothing) | A description for the workspace. |

> | `data.attributes.auto_apply`                            | boolean |  `false`  | If `true`, HCP Terraform will automatically apply changes when a Terraform plan is successful, with some exceptions. |

Reviewer: ```suggestion
| `data.attributes.auto_apply`                            | boolean |  `false`  | If `true`, HCP Terraform automatically applies changes when a Terraform plan is successful. |
```
Unless we can describe the exceptions or link to at least one example, I can see this being frustrating for users.

      }
    },
    "links": {
      "self": "/api/v2/organizations/my-organization/workspaces/no-code-workspace",
      "self-html": "/app/my-organization/workspaces/no-code-workspace"
    }
  }
}
```

## Initiate a no-code workspace update

Upgrading a workspace's no-code provisioning settings is a multi-step process. 

> 1. Use this API to initiate the update. HCP Terraform will start a new plan, which describes the resources that will be added, updated, or removed from the workspace.

Reviewer: ```suggestion
1. Use this API to initiate the update. HCP Terraform starts a new plan, which describes the resources to add, update, or remove from the workspace.
```

| `:workspace_id`      | The ID of workspace.           |
| `:id`                | The ID of update.              |

> Returns the details of a no-code workspace update run, including the run's current state (such as `pending`, `fetching`, `planning`, `planned`, or `cost_estimated`). Refer to [Run States and Stages](https://developer.hashicorp.com/terraform/enterprise/run/states) for more information on the states a run can return.

Reviewer: ```suggestion
Returns the details of a no-code workspace update run, including the run's current state, such as `pending`, `fetching`, `planning`, `planned`, or `cost_estimated`. Refer to [Run States and Stages](https://developer.hashicorp.com/terraform/enterprise/run/states) for more information on the states a run can return.
```
Avoid using parentheses unless it's part of a code sample
